src/parser.rs

```diff
@@ -557,6 +557,63 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// parse a group by expr. a group by expr can be one of group sets, roll up, cube, or simple
+    /// expr.
+    fn parse_group_by_expr(&mut self) -> Result<Expr, ParserError> {
+        if self.parse_keywords(&[Keyword::GROUPING, Keyword::SETS]) {
```

Comment:
I double checked at MySql also supports GROUPING, though it seems it supports `WITH ROLLUP` rather than `ROLLUP`,  https://dev.mysql.com/doc/refman/8.0/en/group-by-modifiers.html 

but seemingly not CUBE https://dev.mysql.com/doc/search/?d=201&p=1&q=cube 

I think we should make parsing this part of the syntax conditional on the the postgresql dialect

Reply:
done now